github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/pipeline/SetCommitStatusExecution.java

```diff
@@ -0,0 +1,123 @@
+package org.jenkinsci.plugins.github.pullrequest.pipeline;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.jenkinsci.plugins.github.pullrequest.utils.ObjectsUtil.isNull;
+import static org.jenkinsci.plugins.github.pullrequest.utils.ObjectsUtil.nonNull;
+
+import org.jenkinsci.plugins.github.pullrequest.GitHubPRCause;
+import org.jenkinsci.plugins.github.pullrequest.GitHubPRTrigger;
+import org.jenkinsci.plugins.github.util.JobInfoHelpers;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.github.GHRepository;
+
+import com.coravy.hudson.plugins.github.GithubProjectProperty;
+import com.google.inject.Inject;
+
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import jenkins.model.JenkinsLocationConfiguration;
+
+/**
+ * Pipeline DSL step to update a GitHub commit status for a pull request.
+ */
+public class SetCommitStatusExecution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+
+    /** YYYYMMDD */
+    private static final long serialVersionUID = 1L;
+
+    @StepContextParameter
+    private transient Run<?, ?> run;
+
+    @StepContextParameter
+    private transient TaskListener log;
+
+    @Inject
+    private transient SetCommitStatusStep config;
+
+    @Override
+    protected Void run() throws Exception {
+        //
+        // Figure out which GitHub repository to send the request to
+        //
+        checkArgument(nonNull(config.getState()), "Missing required parameter 'state'");
+
+        GHRepository repository = resolveRepository();
+        String statusContext = resolveContext();
+
+        final GitHubPRCause cause = run.getCause(GitHubPRCause.class);
+        if (isNull(cause)) {
+            throw new ProjectConfigurationException("pullRequestSetCommitStatus "
+                                + "requires build to be triggered by GitHub Pull Request");
+        }
+        //
+        // Update the commit status
+        //
+        log.getLogger().printf("Setting pull request status %s to %s with message: %s%n",
+                               statusContext,
+                               config.getState(),
+                               config.getMessage());
+        String buildUrl = null;
+        final JenkinsLocationConfiguration globalConfig = JenkinsLocationConfiguration.get();
+        if (nonNull(globalConfig)) {
+            buildUrl = globalConfig.getUrl();
+        }
+        if (isEmpty(buildUrl)) {
+            log.error("Jenkins Location is not configured in system settings. Cannot create a 'details' link.");
```

Comment:
please add slf4j logger and print to LOG and logger. That would help debugging from global jenkins logger.